_posts/2022-08-22-fan-out-and-back-in-using-functions.md

```diff
@@ -0,0 +1,471 @@
+
+In this example we are going to process a CSV file containing URLs to images on Wikipedia. For each URL in the data set we want to run the Inception model and get back image categorizations through machine learning. The inception model can classify what's in a photo based upon a training set made up of objects and animals. As a final processing step we will combine all the individual results into a single output.
+
+![Fan-out/fan-in pattern with OpenFaaS functions](/images/2022-fan-out-and-back-in-using-functions/fan-out-in-example.png)
+> Fan-out/fan-in pattern with OpenFaaS functions
+
+You can find the [full example on GitHub](https://github.com/welteki/openfaas-fan-in-example)
+
+### Fan-out
+S3 will be used as the data store throughout this example. The input files are retrieved from an S3 bucket, results of invocations and the final result will be uploaded to the same bucket. To run this example yourself you will need to create an [Amazon S3 bucket](https://docs.aws.amazon.com/AmazonS3/latest/userguide/GetStartedWithS3.html).
+
+The functions need access to AWS credentials in order to use the bucket. The `faas-cli` can be used to define these secrets.
+```bash
+echo $aws_access_key_id | faas-cli secret create s3-key
+echo $aws_secret_access_key | faas-cli secret create s3-secret
+```
+
+> You can checkout the documentation for more info on [how to use secrets within your functions](https://docs.openfaas.com/reference/secrets/).
+
+The first part of the workflow will consist of two functions. The `creat-batch` function is responsible for initializing the workflow. It accepts the name of a CSV file stored in an S3 bucket as input. The function will retrieve the CSV file containing image URLs and invoke the second function, `run-model` for each URL in the file.
+
+The `run-model` function is responsible for calling the machine learning model, in this case the `inception` function, and uploading the result to S3.
+
+We will use the `python3-http` template for both functions:
+
+```bash
+# Get the python3-flask template from the store
+faas-cli template store pull python3-flask
+
+# Scaffold the functions
+faas-cli new create-batch --lang python3-http-debian
+mv create-batch.yml stack.yml
+faas-cli new run-model --lang python3-http -f stack.yml
+```
+
+Note that we use `python3-http-debian` for the `create-batch` function. [pandas](https://pypi.org/project/pandas/) will be used to process the CSV file. The pandas pip module requires a native build toolchain. It is advisable to use the debian version of the template for native dependencies.
```

Comment:
outside of back ticks should Debian be capitalised?

Reply:
Did you include any native dependencies that meant you switched to this template? If so, "such as .. which I used here"